ext/vulnsrc/rocky/rocky.go

```diff
@@ -0,0 +1,180 @@
+// Package rocky implements a vulnerability source updater using
+// RLSA (Rocky Linux Security Advisories).
+
+package rocky
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"regexp"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/stackrox/scanner/database"
+	"github.com/stackrox/scanner/ext/versionfmt/rpm"
+	"github.com/stackrox/scanner/ext/vulnsrc"
+	"github.com/stackrox/scanner/pkg/commonerr"
+	"github.com/stackrox/scanner/pkg/httputil"
+)
+
+const (
+	url        = "https://errata.rockylinux.org/api/advisories"
+	linkFormat = "https://errata.rockylinux.org/%s"
+)
+
+var (
+	rpmRegexp        = regexp.MustCompile(`(?P<name>.*)-(?P<version>[^\-]+\-[^\-]+)\.(?P<basearch>[^.]+)\.rpm$`)
+	rpmReleaseRegexp = regexp.MustCompile(`.*\.el(?P<version>[\d]+)`)
+)
+
+type jsonData struct {
+	Advisories []jsonAdvisory `json"advisories"`
+}
+
+type jsonAdvisory struct {
+	Type             string   `json:"type"`
+	Name             string   `json:"name"`
+	Description      string   `json:"description"`
+	AffectedProducts []string `json:"affectedProducts"`
+	Severity         string   `json:"severity"`
+	CVEs             []string `json:"cves"`
+	RPMs             []string `json:"rpms"`
+}
+
+type updater struct{}
+
+func init() {
+	vulnsrc.RegisterUpdater("rocky", &updater{})
+}
+
+func (u *updater) Update(datastore vulnsrc.DataStore) (resp vulnsrc.UpdateResponse, err error) {
+	log.WithField("package", "Rocky").Info("Start fetching vulnerabilites")
+
+	// Download JSON
+	r, err := httputil.GetWithUserAgent(url)
+	if err != nil {
+		log.WithError(err).Error("could now download Rocky's update")
+		return resp, commonerr.ErrCouldNotDownload
+	}
+	defer r.Body.Close()
+
+	if !httputil.Status2xx(r) {
+		log.WithField("StatusCode", r.StatusCode).Error("Failed to update Rocky")
+		return resp, commonerr.ErrCouldNotDownload
+	}
+
+	// Parse JSON
+	resp, err = buildResponse(r.Body)
+	if err != nil {
+		return resp, err
+	}
+
+	return resp, nil
+}
+
+func (u *updater) Clean() {}
+
+func buildResponse(jsonReader io.Reader) (resp vulnsrc.UpdateResponse, err error) {
+	// Unmarshal JSON
+	var data jsonData
+	err = json.NewDecoder(jsonReader).Decode(&data)
+	if err != nil {
+		log.WithError(err).Error("could not unmarshal Rocky's JSON")
+		return resp, commonerr.ErrCouldNotParse
+	}
+
+	// Extract vulnerability data from Rocky's JSON schema.
+	resp.Vulnerabilities = parseRockyJSON(&data)
+
+	return resp, nil
+}
+
+func parseRockyJSON(data *jsonData) (vulnerabilities []database.Vulnerability) {
+	vulns := make(map[string]*database.Vulnerability)
+
+	for _, advisory := range data.Advisories {
+		if !strings.EqualFold(advisory.Type, "security") {
+			continue
+		}
+
+		vulnName := advisory.Name
+		vulnerability, vulnerabilityAlreadyExists := vulns[vulnName]
+
+		if !vulnerabilityAlreadyExists {
```

Comment:
what does it mean if I see two of the same vuln ID?

Reply:
This shouldn't happen, this is likely something copied over from another vulnsrc updater. I've removed it.